sympy/series/fourier.py

```diff
@@ -90,6 +91,37 @@ def _find_x(func):
     return sympify((x, start, stop))
 
 
+def finite_check(f, x):
+    def check_fx(exprs, x):
+        return x not in exprs.free_symbols
```

Comment:
If x isn't in the expression then it's just a constant factor, no?

Reply:
The expression can consist of multiple variables. I am checking for a form ```f(y)sin(a*x+b)``` where ```f(y)``` can be constant or an expression of other variables.